### PR TITLE
Avoid layout shift when showing the Setup CTA Banner

### PR DESCRIPTION
**Pull request by ankitrox**
## Summary

Addresses issue:

- #8707 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js [hunk 1]
```diff
@@ -71,24 +66,15 @@ function AudienceSegmentationSetupCTAWidget( { Widget } ) {
...
+	const analyticsIsDataAvailableOnLoad = useSelect( ( select ) => {
+		select( MODULES_ANALYTICS_4 ).isGatheringData();
+		return select( MODULES_ANALYTICS_4 ).isDataAvailableOnLoad();
 	} );
 
-	if ( configuredAudiences !== null || ! hasDataWithinPast90Days ) {
+	if (
+		( configuredAudiences && configuredAudiences.length ) ||
+		! analyticsIsDataAvailableOnLoad
+	) {
```
Comment (hussain-t) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js:77]: This doesn't check for `undefined` but only for the truthy value, and the array is not empty.

**Review by hussain-t — CHANGES_REQUESTED** [review]
Thanks, @ankitrox. The changes look good overall. However, there are merge conflicts due to the changes in [8133](https://github.com/google/site-kit-wp/issues/8133), which also adds tests for the `AudienceSegmentationSetupCTAWidget` component. Please resolve these conflicts while ensuring that the existing tests are maintained.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js (hunk 1 above)
Comment (ankitrox) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js]: @hussain-t This is checking if `configuredAudiences` are present. If yes, then we do not need to render the CTA widget as we already have setup the audiences.

Thoughts?

**Comment by ankitrox** [comment]
@hussain-t Thank you for reviewing the PR. I have resolved the conflicts in the PR and also added response to your [comment](https://github.com/google/site-kit-wp/pull/8759#discussion_r1617025939).

Can you please re-review this PR? Assigning this to you.

Thanks

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.test.js [hunk 2]
```diff
@@ -276,5 +277,111 @@ describe( 'AudienceSegmentationSetupCTAWidget', () => {
...
+			await waitForRegistry();
+
+			expect( () =>
+				getByText(
+					'Learn how different types of visitors interact with your site'
+				)
+			).toThrow();
+		} );
+
+		it( 'banner is not visible when configured audiences present and google analytics data loaded on the page', async () => {
```
Comment (hussain-t) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.test.js:351]: Can we update the test case title to match the existing ones?
```suggestion
		it( 'should not render the widget when no configured audiences and google analytics data is not loaded on the page', async () => {
```

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js (hunk 1 above)
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js]: Makes sense 👍

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js [hunk 3]
```diff
@@ -153,8 +138,8 @@ function AudienceSegmentationSetupCTAWidget( { Widget, WidgetNull } ) {
 	}
 
 	if (
-		configuredAudiences !== null ||
-		! hasDataWithinPast90Days ||
+		( configuredAudiences && configuredAudiences.length ) ||
```
Comment (hussain-t) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.js:141]: Let's simplify it
```suggestion
		configuredAudiences?.length ||
```

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.test.js [hunk 4]
```diff
@@ -276,5 +277,111 @@ describe( 'AudienceSegmentationSetupCTAWidget', () => {
...
+			);
+
+			// Wait for resolvers to finish to avoid an unhandled React state update.
+			await waitForRegistry();
+
+			expect( () =>
+				getByText(
+					'Learn how different types of visitors interact with your site'
+				)
+			).toThrow();
```
Comment (hussain-t) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.test.js:348]: Can we use `queryByText` and `not.toBeInTheDocument`?

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.test.js [hunk 5]
```diff
@@ -276,5 +277,111 @@ describe( 'AudienceSegmentationSetupCTAWidget', () => {
...
```
Comment (hussain-t) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.test.js:384]: Can we use `queryByText` and `not.toBeInTheDocument`?

**Review by hussain-t — CHANGES_REQUESTED** [review]
Thanks, @ankitrox. The changes look good overall. However, I left a few nit-picky feedbacks. I will address them myself to avoid another iteration.

**Review by hussain-t — APPROVED** [review]
Excellent work, @ankitrox. LGTM 🎖️ 

I've verified that the build failures are unrelated to this PR.

**Review by nfmohit — CHANGES_REQUESTED** [review]
Excellent work here, thank you @ankitrox.

I have left some minimal and minor feedback, mostly around language and conventions. Please let me know what you think, thank you!

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.test.js [hunk 6]
```diff
@@ -122,6 +123,112 @@ describe( 'AudienceSegmentationSetupCTAWidget', () => {
...
+			const settings = {
+				configuredAudiences: [
+					audiencesFixture[ 0 ],
+					audiencesFixture[ 1 ],
+					audiencesFixture[ 2 ],
+				],
+				isAudienceSegmentationWidgetHidden: false,
+			};
+
+			// Set the data available on page load to true.
```
Comment (nfmohit) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget.test.js:207]: Looks like this got left out. As this is a very minor change, I'll apply it myself. Thanks!

**Review by nfmohit — CHANGES_REQUESTED** [review]
Thank you for addressing the feedback, @ankitrox ! I found one remaining instance of a comment where the language could be improved which I've mentioned as part of this review, but as that is very minor, I've applied the fix myself. I hope that is okay with you. Thanks!

**Review by nfmohit — APPROVED** [review]
LGTM, thank you @ankitrox !

**Pull request merged**